walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,136 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.Set;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.Iterator;
+
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+    private ArrayList<TimeRange> availableTimes = new ArrayList<>();
+
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+        availableTimes.clear();
+
+        // If the request is too long, return empty ArrayList
+        if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+            return availableTimes;
+        }
+
+        /* Any time is available during the start. Time ranges will 
+            be dynamically added and removed from the availableTimes 
+            array as the program loops through the events */
+        availableTimes.add(TimeRange.WHOLE_DAY);
+        
+        // If the set of attendees is empty, the entire day is available
+        if (request.getAttendees().isEmpty() && request.getOptionalAttendees().isEmpty()) {
+            return availableTimes;
+        }
+
+        Collection<String> attendees = request.getAttendees();
+        Collection<String> optionalAttendees = request.getOptionalAttendees();
+
+        // Boolean: mandatoryAttendee
+        Map<String, Boolean> attendeesMap = new HashMap<String, Boolean>();
+        for (String attendee : attendees) { 
+            attendeesMap.put(attendee, true); 
+        }
+        for (String attendee : optionalAttendees) { 
+            attendeesMap.put(attendee, false); 
+        }
```

Comment:
This is a good idea, but I would suggest naming your `attendeesMap` with a more descriptive name.